plugins/modules/swim_workflow_manager.py

```diff
@@ -565,23 +617,54 @@ def site_exists(self, site_name):
         site_exists = False
         site_id = None
         response = None
+
         try:
-            response = self.dnac._exec(
-                family="sites",
-                function='get_site',
-                op_modifies=True,
-                params={"name": site_name},
-            )
+            if self.dnac_version <= self.version_2_3_5_3:
+                response = self.dnac._exec(
+                    family="sites",
+                    function='get_site',
+                    op_modifies=True,
+                    params={"name": site_name},
+                )
+                if response:
+                    self.log("Received API response from 'get_site': {0}".format(str(response)), "DEBUG")
+                    site = response.get("response")
+                    site_id = site[0].get("id")
+                    site_exists = True
+
         except Exception as e:
-            self.msg = "An exception occurred: Site '{0}' does not exist in the Cisco Catalyst Center".format(site_name)
+            self.msg = (
+                "'An exception occurred: Site '{0}' does not exist in the Cisco Catalyst Center' or 'In this version '{1}'"
+                ", 'get_site' is not supported'".format(site_name, self.payload.get("dnac_version"))
+            )
             self.log(self.msg, "ERROR")
             self.module.fail_json(msg=self.msg)
 
-        if response:
-            self.log("Received API response from 'get_site': {0}".format(str(response)), "DEBUG")
-            site = response.get("response")
-            site_id = site[0].get("id")
-            site_exists = True
+        try:
+            if self.dnac_version >= self.version_2_3_7_6:
+                self.log("inside if block, starting responce")
+                response = self.dnac._exec(
+                    family="site_design",
+                    function='get_sites',
+                    op_modifies=True,
+                    params={"name_hierarchy": site_name},
+                )
+                self.log("site response")
+                self.log(response)
+
+                if response:
+                    self.log("Received API response from 'get_sites': {0}".format(str(response)), "DEBUG")
+                    site = response.get("response")
+                    site_id = site[0].get("id")
+                    site_exists = True
+
+        except Exception as e:
+            self.msg = (
+                "'An exception occurred: Site '{0}' does not exist in the Cisco Catalyst Center' or 'In this version '{1}'"
+                ", 'get_sites' is not supported'".format(site_name, self.payload.get("dnac_version"))
+            )
+            self.log(self.msg, "ERROR")
+            self.module.fail_json(msg=self.msg)
```

Comment:
avoid using the fail_json as it will not give the correct message in the console instead of this set the self status to failed and check the return status here only it'll stop the execution of module.
```
self.status = "failed"
self.msg = (
            "'An exception occurred: Site '{0}' does not exist in the Cisco Catalyst Center' or 'In this version '{1}' "
            ", 'get_sites' is not supported'"
).format(site_name, self.payload.get("dnac_version"))
self.log(self.msg, "ERROR")
self.check_return_status()

Reply:
updated the code